SwiftDraft/ReliantFramework/ReliantFrameworkTests/ReliantFrameworkTests.swift

```diff
@@ -54,17 +54,25 @@ struct ContextNeedingContext : ReliantContext {
         return ContextNeedingContext()
     }
 }
+class SubWaver : Waver {
+    func wave(reason: String) -> String {
+        return "Substitute waving"
+    }
+}
 
-
-
-
+class SubstituteContext : SimpleReferenceContext {
+    override init() {
+        super.init()
+        waver = SubWaver()
+    }
+}
 
 class ReliantFrameworkTests: XCTestCase {
     
     override func setUp() {
         super.setUp()
         ReliantFrameworkTestsHelper.sharedInsance.reset()
-        ContextCache.sharedInstance.cache = Dictionary()
+        ContextCache.standard.removeAll()
```

Comment:
We'll always have global state, I think, even if we opt to bind the the `relyOn` function to some other object, this object will either be a framework object like an app delegate, or something else that sticks around longer than necessary.